source/components/IDInput/index.jsx

```diff
@@ -80,6 +82,17 @@ const IDInput = ({
     setContactName(e.target.value);
   };
 
+  const searchContactFromId = (e) => {
+    onChange(e.target.value);
+    contacts.forEach((contact) => {
+      contact.contacts.forEach((i) => {
+        if (e.target.value === i.id) {
+          setSelectedContact(i);
+        }
+      });
```

Comment:
this is weird 🤔  do we have a `contacts` field on each `contact` object? 
besides that you could do 
```
const id = e.target.value;
onChange(id);
const allContacts = contacts.flatMap((contact) => contact.contacts);
setSelectedContact(allContacts.find((contact) => contact.id === id));
```